wordpress-mobile/WordPress-Android
# Split ThumbnailViewHolder.kt into 2 child view holders

This PR prepares the photo picker adapter and the thumbnail view holder for more types (like documents/audios) in the future. This is done by splitting the photo picker item into 2 subtypes and creating 2 view holders to keep the items. 
As part of this PR I've also removed the animation on the selection count (which was added in a previous PR and is not in the released version). 

If you have time, it would be great if you could review this PR @khaykov, thanks!

To test:
- Go to a Gutenberg post
- Add an image block
- Click on "Add media" and on "Choose from device"
- Compare the grid view of images to the production version
- Click on an item
- Check that it's correctly selected
- Leave action mode
- The item is correctly unselected


To test:
- Go to a Gutenberg post
- Add a Gallery block
- Click on "Add media" and on "Choose from device"
- Compare the grid view of images to the production version
- Click on some items item
- Check that they're correctly selected
- The animation looks as expected
- Leave action mode
- The items are correctly unselected


To test:
- Add a story
- Check that the videos and images in the list look correct
- Try selecting/unselecting items
- The animation is correct
- You can still create a story

PR submission checklist:
